core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -207,11 +210,28 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
       }
     }
     catch {
-      case NonFatal(e) => log.warn(e.getMessage, e)
+      case NonFatal(e) => {
+        log.warn("Exception occurred while collecting External Table Statistics: "
+            + e.getMessage, e)
+      }
     }
 
     if(result.flatMap(_.getRegionStats.asScala).size == 0) {
-      // Return last updated tableSizeInfo
+      // Return last successfully updated tableSizeInfo
+      // Check for if any table is present in catalog
+      val allStoreTables = Misc.getMemStore.getExternalCatalog.getAllStoreTablesInCatalog(true)
+      val allTablesList = scala.collection.mutable.Buffer.empty[String]
+      allStoreTables.asScala.keys.foreach(schema => {
+        allTablesList ++= allStoreTables.get(schema).asScala.map(tbl => {
+          schema.concat(".").concat(tbl)
+        })
+      })
+
+      if(allTablesList.size == 0 && tableSizeInfo.size > 0){
+        // No table is present in any db schema, reset tableSizeInfo
+        tableSizeInfo = Map.empty[String, SnappyRegionStats]
+      }
+
```

Comment:
Please fix the formatting. In fact in most of your checkins. There should be space between "if" and opening parenthesis. "if()". Please use "Project" code settings with codeStyleSettings.xml copied to .idea as noted in build docs. IDEA >= 16 must be showing this in red since this violates code style guidelines.

Reply:
Code formatted as suggested.

I have followed snappydata documentation while Setting up IntelliJ IDEA and also have added codeStyleSettings.xml into .idea directory. But I never get any code styling error automatically. If I reformat code then it uses the specified code style codeStyleSettings.xml. Therefore I always do it manually before committing the code.